### PR TITLE
[Feature] Refactor Pod and containter creator part2

### PR DESCRIPTION
**Pull request by informalict**
add 4 new functions to the interfaces:
GetArgs
GetName
GetVolume
GetVolumeMounts

### CODE REVIEW

File: pkg/deployment/resources/pod_creator_arangod.go [hunk 1]
```diff
@@ -66,7 +66,8 @@ type MemberArangoDPod struct {
 	resources        *Resources
 	imageInfo        api.ImageInfo
 	autoUpgrade      bool
-	id               string
+	args             []string
```
Comment (ajanikow) [inline pkg/deployment/resources/pod_creator_arangod.go:69]: Why providing it as static list? We can generate it

File: pkg/deployment/resources/pod_creator_arangod.go [hunk 2]
```diff
@@ -66,7 +66,8 @@ type MemberArangoDPod struct {
...
+	volumes          pod.Volumes
```
Comment (ajanikow) [inline pkg/deployment/resources/pod_creator_arangod.go:70]: Also here, I would vote to generate it on demand

**Review by ajanikow — CHANGES_REQUESTED** [review]

File: pkg/deployment/resources/pod_creator_arangod.go (hunk 1 above)
Comment (informalict) [reply pkg/deployment/resources/pod_creator_arangod.go]: changed

File: pkg/deployment/resources/pod_creator_arangod.go (hunk 2 above)
Comment (informalict) [reply pkg/deployment/resources/pod_creator_arangod.go]: changed

**Review by ajanikow — APPROVED** [review]

**Pull request merged**